hydrator-common/src/main/java/co/cask/hydrator/common/batch/ConfigurationUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.common.batch;
+
+import com.google.common.collect.MapDifference;
+import com.google.common.collect.Maps;
+import org.apache.hadoop.conf.Configuration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility class for providing operations on the {@link Configuration}.
+ */
+public final class ConfigurationUtils {
+  private ConfigurationUtils() {
+    // no-op
+  }
+
+  /**
+   * Get the {@link Map} of properties which are newly added or have updated in the configuration instance passed
+   * to this method as a parameter. The modified configurations are compared with the default configurations
+   * created using {@code new Configuation()}.
+   * @param modifiedConf the updated configurations
+   * @return the {@link Map} of properties which are newly added or have updated in the modifiedConf.
+   */
+  public static Map<String, String> getNonDefaultConfigurations(Configuration modifiedConf) {
+    MapDifference<String, String> mapDifference = Maps.difference(getConfigurationAsMap(new Configuration()),
```

Comment:
I think its safer to make this a non-utils class that takes an original Configuration in its constructor and clones it, then uses that original instead of new Configuration() here.